editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/CamelFacetVersionComparator.java

```diff
@@ -0,0 +1,35 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.fusesource.ide.projecttemplates.util;
+
+import java.util.Comparator;
+
+import org.osgi.framework.Version;
+
+/**
+ * basically an osgi version comparator
+ * 
+ * @author lhein
+ */
+public class CamelFacetVersionComparator implements Comparator {
```

Comment:
we can use Generics:

public class CamelFacetVersionComparator implements Comparator<String> {

	@Override
	public int compare(String v0, String v1) {
		Version cv0 = new Version(v0);
		Version cv1 = new Version(v1);
		return cv0.compareTo(cv1);
	}